### PR TITLE
6/api/reviews/review_id/comments

### PR DESCRIPTION
**Pull request by JoshFost**
believe I have finished the task

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -148,9 +148,57 @@ describe("app", () => {
...
+            "created_at",
+            expect.any(String)
+          );
+          expect(body.comments[0]).toHaveProperty("author", expect.any(String));
+          expect(body.comments[0]).toHaveProperty("body", expect.any(String));
+          expect(body.comments[0]).toHaveProperty(
+            "review_id",
+            expect.any(Number)
+          );
+        });
```
Comment (johnhutcheon) [inline __tests__/app.test.js]: Rather than checking [0] here, could we please iterate through to check each element in the array with a forEach?

File: __tests__/app.test.js [hunk 2]
```diff
@@ -148,9 +148,57 @@ describe("app", () => {
...
+    it("400:should responds with an error message when an invalid path is used", () => {
+      return request(app)
+        .get("/api/reviews/not-a-number/comments")
+        .expect(400)
+        .then(({ body }) => {
+          expect(body).toEqual({ msg: "Bad Request" });
+        });
+    });
+  });
 });
```
Comment (johnhutcheon) [inline __tests__/app.test.js:200]: Can we add one more 200 test for a valid ID with no comments? This should respond with an empty array of comments

**Review by johnhutcheon — CHANGES_REQUESTED** [review]
Hey Josh, could you just see my comments and make the amendments then resubmit please? Thanks!

File: __tests__/app.test.js [hunk 3]
```diff
@@ -148,9 +148,53 @@ describe("app", () => {
...
         .expect(404)
         .then(({ body }) => {
-          console.log(body.msg);
           expect(body.msg).toBe("Not Found");
         });
     });
   });
 });
+describe("/api/reviews/:review_id/comments", () => {
+  it("200: GET: it should respond with an object of comments for the given review_id", () => {
```
Comment (jimstev2447) [inline __tests__/app.test.js]: Do you want an object of comments or an Array of comments?

File: __tests__/app.test.js [hunk 4]
```diff
@@ -148,9 +148,53 @@ describe("app", () => {
...
+    return request(app)
+      .get("/api/reviews/1/comments")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.comments).toBeInstanceOf(Object);
+      });
+  });
+  it("200:GET: should respond with an empty array if review_id has no comments", () => {
+    const reviewId = 100;
```
Comment (jimstev2447) [inline __tests__/app.test.js]: Does this review_id exist? if not then I would expect this to 404, 
What we want to get from this is that 
- if the review exists but doesnt have any comments we want a 200 and an empty array
- If the review doesn't exist then it should 404

File: __tests__/app.test.js [hunk 5]
```diff
@@ -148,9 +148,53 @@ describe("app", () => {
...
+    return request(app)
+      .get(`/api/reviews/${reviewId}/comments`)
+      .expect(200)
+      .then(({ body }) => {
+        console.log(body);
```
Comment (jimstev2447) [inline __tests__/app.test.js]: please be careful about leaving in console logs inn what is you finished code

File: __tests__/app.test.js [hunk 6]
```diff
@@ -148,9 +148,53 @@ describe("app", () => {
...
+          expect(body).toEqual({ comments: [] });
+        }
+      });
+  });
+  it("200:GET: should respond with the correct properties", () => {
+    return request(app)
+      .get("/api/reviews/1/comments")
+      .expect(200)
+      .then(({ body }) => {
+        body.comments.forEach((comment) => {
```
Comment (jimstev2447) [inline __tests__/app.test.js:182]: review_id 1 doesnt actually have any comments, which is sending back an empty array and because the array is empty this forEach is never being invoked, so this tests isn't running

File: models/categoryDataModel.js [hunk 7]
```diff
@@ -58,3 +58,27 @@ exports.fetchReviewById = (review_id, sort_by) => {
       }
     });
 };
+
+exports.fetchCommentsByReviewId = (review_id, sort_by) => {
+  if (isNaN(review_id)) {
+    return Promise.reject({ status: 400, msg: "Bad Request" });
```
Comment (jimstev2447) [inline models/categoryDataModel.js:64]: I think if you send this to the DB without checking here the db will reject and you can handle this by adding a function to  your error handling middleware chain
